packages/flutter/lib/src/cupertino/date_picker.dart

```diff
@@ -0,0 +1,358 @@
+      itemExtent: _kItemExtent,
+      backgroundColor: _kBackgroundColor,
+      onSelectedItemChanged: (int index) {
+        setState(() {
+          selectedSecond = index;
+          widget.onTimerDurationChanged(
+            new Duration(
+              hours: selectedHour,
+              minutes: selectedMinute,
+              seconds: selectedSecond));
+        });
+      },
+      children: new List<Widget>.generate(60 ~/ widget.secondInterval, (int index) {
+        final int second = index * widget.secondInterval;
+        final String semanticsLabel = textDirectionFactor == 1
+          ? localizations.timerPickerSecond(second) + localizations.timerPickerSecondLabel(second)
+          : localizations.timerPickerSecondLabel(second) + localizations.timerPickerSecond(second);
+
+        return Semantics(
+          label: semanticsLabel,
+          child: Container(
+            alignment: Alignment(-1.0 * textDirectionFactor, 0.0),
+            child: Container(
+              alignment: Alignment(1.0 * textDirectionFactor, 0.0),
+              // Adds some spaces between words.
+              padding: const EdgeInsets.symmetric(horizontal: 2.0),
+              width: _kPickerWidth / 6,
+              child: Text(localizations.timerPickerSecond(second)),
+            ),
+          ),
+        );
+      }),
+    );
+  }
+
+  Widget _buildSecondColumn(BuildContext context) {
+    final int textDirectionFactor =
+      Directionality.of(context) == TextDirection.ltr ? 1 : -1;
+
+    final CupertinoLocalizations localizations =
+      CupertinoLocalizations.of(context) ?? const DefaultCupertinoLocalizations();
+
+    return Stack(
+      children: <Widget>[
+        _buildSecondPicker(context),
+        // The second label.
+        IgnorePointer(
+          child: Container(
+            alignment: Alignment(-1.0 * textDirectionFactor, 0.0),
+            padding: textDirectionFactor == 1
+              ? const EdgeInsets.only(left: _kPickerWidth / 6)
+              : const EdgeInsets.only(right: _kPickerWidth / 6),
+            child: Container(
+              alignment: Alignment(-1.0 * textDirectionFactor, 0.0),
+              // Adds some spaces between words.
+              padding: const EdgeInsets.symmetric(horizontal: 2.0),
+              child: _buildLabel(localizations.timerPickerSecondLabel(selectedSecond)),
+            ),
+          ),
+        ),
+      ],
+    );
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    // The timer picker can be divided into 3 columns corresponding to hour,
+    // minute, and second. Each column consists of a scrollable and a fixed
+    // label on top of it.
+
+    return new MediaQuery(
+      data: const MediaQueryData(
+        // The native iOS picker's text scaling is fixed, so we will also fix it
+        // as well in our picker.
+        textScaleFactor: 1.0,
+      ),
+      child: new Row(
+        children: <Widget>[
+          new Expanded(
```

Comment:
Why not just evenly distribute them instead?

Reply:
The picker's width is fixed, so distributing evenly will make the picker stretch out to fit in the spaces.